ReactSkia/sdk/CurlNetworking.cpp

```diff
@@ -116,11 +114,18 @@ void CurlNetworking::processNetworkRequest(CURLM *curlMultiHandle) {
             (curlRequest->curlResponse->responseTimeout = true)
             :(curlRequest->curlResponse->responseTimeout = false);
         curl_multi_remove_handle(curlMultiHandle, curlHandle);
-        if(curlRequest->shouldCacheData()) {
+        if(msg->data.result == CURLE_OK && curlRequest->shouldCacheData()) {
           if(networkCache_->isAvailableInCache(curlRequest->URL)) {
             RNS_LOG_DEBUG("Data is already in cache");
           } else {
-            networkCache_->setCache(curlRequest->URL, curlRequest->curlResponse);
+            double downloadedSize;
+            curl_easy_getinfo(curlHandle, CURLINFO_SIZE_DOWNLOAD, &downloadedSize);
+            if(!networkCache_->needEvict(downloadedSize)) {
```

Comment:
pass (curlRequest->curlResponse->contentSize + headbufferSize) in needevict method